File: java/observer/NewsLetter/observer/Subscriber.java
package observer;

import subject.Newsletter;

public class Subscriber implements Observer {

    private String name;
    private Newsletter newsletter;

    public Subscriber(Newsletter newsletter) {
        this.newsletter = newsletter;
        newsletter.registerObserver(this);
    }

    @Override
    public void update(String name) {
        this.name = name;
        display();
    }

    private void display() {
        System.out.println("New Subscriber: +" + name + " added to newsletter");
    }
}

File: javascript/observer/NewsLetter/observer/Subscriber.js
import {Observer} from "./Observer.js";
import {Newsletter} from "../subject/Newsletter.js";

export class Subscriber extends Observer {

    #name
    #newsletter = new Newsletter()

    constructor(newsLetter) {
        super();
        this.#newsletter = newsLetter
        this.#newsletter.registerObserver(this)
    }

    update(name) {
        super.update(name);
        this.#name = name
        this.display()
    }

    display() {
        console.log("New Subscriber: +" + this.#name + " added to the newsletter")
    }

}